File: Mentor.Me.Web/ClientApp/src/app/shared/types/assignment.ts
import AssignmentStatus from './enums/task-status.enum';
import User from './user';

export default class Assignment {
    public id: string;
    public name: string;
    public description: string;
    public deadLine: Date;
    public ownerId: string;
    public skillId: string;
    public status: AssignmentStatus;
    public members: User[];
}

File: Mentor.Me.Data/Entities/Assignment.cs
using System;
using System.Collections.Generic;
namespace Mentor.Me.Data.Entities
{
    public class Assignment : IEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Deadline { get; set; }
        public Guid OwnerId { get; set; }
        public Guid SkillId { get; set; }
        public AssignmentStatus Status { get; set; }
        public IEnumerable<User> Members { get; set; }
    }
}